src/controller/java/JniReferences.h

```diff
@@ -17,17 +17,21 @@
 
 #pragma once
 
-#include <jni.h>
-
 #include <core/CHIPError.h>
-
+#include <jni.h>
+#include <pthread.h>
 namespace {
-JavaVM * sJvm = nullptr;
+jclass clusterExceptionCls;
+JavaVM * sJvm              = nullptr;
+pthread_mutex_t sStackLock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
The lock shouldn't be in the anonymous namespace. All modules should see exactly the same instance to make the locking work properly.

Reply:
I created a class for `JniReferences` and have the lock as a static member now.